### PR TITLE
chore: add tests for connecting with a profile from an aws credentials file MONGOSH-1609

### PR DESCRIPTION
**Pull request by lerouxb**

### CODE REVIEW

File: packages/cli-repl/src/run.ts [hunk 1]
```diff
@@ -220,6 +221,7 @@ async function main() {
     });
   } catch (e: any) {
     console.error(`${e?.name}: ${e?.message}`);
+    console.error('full error: ' + util.inspect(e, { depth: null }));
```
Comment (addaleax) [inline packages/cli-repl/src/run.ts:224]: Leftover? (Fwiw, I do this a ton too…)

File: packages/e2e-tests/test/e2e-aws.spec.ts [hunk 2]
```diff
@@ -295,4 +298,86 @@ describe('e2e AWS AUTH', function () {
       });
     });
   });
+
+  context('with aws profile', function () {
+    const awsConfigPath = path.join(
+      os.tmpdir(),
+      `aws-config-${new Date().getTime()}-${Math.random()}`
+    );
```
Comment (addaleax) [inline packages/e2e-tests/test/e2e-aws.spec.ts:306]: `useTmpdir()` from `./repl-helpers.ts`?

File: packages/cli-repl/src/run.ts (hunk 1 above)
Comment (lerouxb) [reply packages/cli-repl/src/run.ts:224]: yeah. There's nothing more there anyway ;)

File: packages/e2e-tests/test/e2e-aws.spec.ts (hunk 2 above)
Comment (lerouxb) [reply packages/e2e-tests/test/e2e-aws.spec.ts:306]: Didn't know about that. copy/pasted this from another test.

File: packages/cli-repl/src/run.ts (hunk 1 above)
Comment (lerouxb) [reply packages/cli-repl/src/run.ts:224]: Would be nice if we could access the most recent error from the test somehow.

**Comment by lerouxb** [comment]
I suppose we can always re-open this if we ever need it.

**Pull request closed without merging**